apps/project/serializers.py

```diff
@@ -128,10 +128,12 @@ class ProjectMembershipSerializer(RemoveNullFieldsMixin,
         read_only=True,
         many=True,
     )
+    role_title = serializers.CharField(source='role.title', read_only=True)
```

Comment:
Let's define SimpleRoleSerializer and use that instead.

role_details = SimpleRoleSerializer(source='role', read_only=True)

Make sure to also use select related. So role_id and role_title should be enough.